captum/metrics/_core/infidelity.py

```diff
@@ -0,0 +1,314 @@
+    _format_tensor_into_tuples,
+    _is_tuple,
+    _run_forward,
+)
+
+
+def infidelity_attr(
+    forward_func,
+    perturb_func,
+    inputs,
+    attributions,
+    additional_forward_args=None,
+    target=None,
+    n_samples=10,
+    max_examples_per_batch=None,
+):
+    r"""
+    Explanation infidelity represents the expected mean-squared error
+    between the explanation multiplied by a meaningful input perturbation
+    and the differences between the predictor function at its input
+    and perturbed input.
+    More details about the measure can be found in the following paper:
+    https://arxiv.org/pdf/1901.09392.pdf
+
+    It is derived from the completeness property of well-known attribution
+    algorithms and is an optimal and generalized notion of Sensitivy-n.
+    The latter measures correlations between the sum of the attributions
+    and the differences of the predictor function at its input and
+    fixed baseline. More details about the Sensitivity-n can be found here:
+    https://arxiv.org/pdf/1711.06104.pdfs
+
+    The users can perturb the inputs any desired way by providing any
+    perturbation function that takes the inputs and returns perturbed
+    inputs and corresponding perturbations.
+
+    This specific implementation is primarily tested for attribution-based
+    explanation methods but the idea can be expanded to use for non
+    attribution-based interpretability methods as well.
+
+    Args:
+
+        forward_func (callable):
+                The forward function of our model or any modification of it.
+        perturb_func (callable):
+                The perturbation function of model inputs. This function takes
+                model inputs as an argument and returns a tuple of perturbations
+                and perturbeded inputs. If there is more than one input,
+                corresponding perturbations must be computed and returned as tuples
+                in the following format:
+                    (perturb1, perturb2, ... perturbN), (input1, input2, ... inputN)
+                In case inputs is a single tensor, the function can return a tuple
+                of perturbations and inputs such as:
+                    perturb, inputs
+
+        inputs (tensor or tuple of tensors):  Input for which
+                attributions are computed. If forward_func takes a single
+                tensor as input, a single input tensor should be provided.
+                If forward_func takes multiple tensors as input, a tuple
+                of the input tensors should be provided. It is assumed
+                that for all given input tensors, dimension 0 corresponds
+                to the number of examples (aka batch size), and if
+                multiple input tensors are provided, the examples must
+                be aligned appropriately.
+
+        attributions (tensor or tuple of tensors):
+                Attribution scores computed based on an attribution algorithm.
```

Comment:
One thing to be careful of with the attributions here is that they should technically be prior to multiplying by (inputs - baselines) for methods that perform this, such as IG and DeepLift. This is because computing the infidelity multiplies the attribution by the perturbation, which essentially assumes that the attribution is expressed in terms of the output change per unit of feature change. This is true for methods such as Saliency, which just take the gradient, but for methods like IG, DeepLift, GradientShap, etc., the attributions should probably be divided by (inputs - baselines) first. 
It would be good to put some explanation / warning to make sure to apply that before passing attributions or even an option which would handle that division internally.

Reply:
To that point they have a subsection called:
`2.5 Local and Global Explanations`
If it is a local explanation then it shouldn't be multiplied by (input - baseline)
In case of global it still holds if `I` is a subset of inputs - baseline
```
Definition 2.1 are naturally local explanations as I is real-valued. However, this can be
easily modified to a global explanation by multiplying with x - x0 when I is a subset of x - x0.
```